frontend/public/components/import-yaml-results.tsx

```diff
@@ -0,0 +1,299 @@
+  const [errors, setErrors] = React.useState(false);
+  const [activeSortIndex, setActiveSortIndex] = React.useState(-1);
+  const [activeSortDirection, setActiveSortDirection] = React.useState(null);
+  const [ready, setReady] = React.useState(false);
+  React.useEffect(() => {
+    // Add a ready state so we're not rendering a flicker of a stale table (with the browser back button)
+    // before the history.goBack action occurs
+    if (location.state.done) {
+      return history.goBack();
+    }
+    setReady(true);
+    const requests = createRequests(location.state.objs, models);
+    Promise.allSettled(requests).then(() => {
+      history.replace({ state: { ...location.state, done: true } });
+      setSending(false);
+    });
+    requests.forEach((resourceRequest, index) => {
+      resourceRequest
+        .then((resourceObj) =>
+          setRows((prevRows) =>
+            // eslint-disable-next-line max-nested-callbacks
+            prevRows.map((resourceRow) =>
+              resourceObj?.metadata?.name === resourceRow.obj.metadata.name &&
+              resourceObj?.kind === resourceRow.obj.kind &&
+              resourceObj?.metadata?.namespace === resourceRow.obj.metadata.namespace
+                ? {
+                    obj: { ...resourceObj },
+                    status: { creating: false, message: t('public~Created') },
+                  }
+                : resourceRow,
+            ),
+          ),
+        )
+        .catch((error) => {
+          setErrors(true);
+          setRows((prevRows) =>
+            // eslint-disable-next-line max-nested-callbacks
+            prevRows.map((resourceRow) =>
+              location.state.objs[index]?.metadata?.name === resourceRow.obj.metadata.name &&
+              location.state.objs[index]?.kind === resourceRow.obj.kind &&
+              location.state.objs[index]?.metadata?.namespace === resourceRow.obj.metadata.namespace
+                ? {
+                    ...resourceRow,
+                    status: {
+                      creating: false,
+                      error: true,
+                      message: t('public~Error: {{error}}', { error: error.message }),
+                    },
+                  }
+                : resourceRow,
+            ),
+          );
+        });
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+  const onSort = (event, index, direction) => {
+    setActiveSortIndex(index);
+    setActiveSortDirection(direction);
+    // sorts the rows
+    const updatedRows = rows.sort((a, b) => {
+      if (direction === 'asc') {
+        return getTableCell(a, index).localeCompare(getTableCell(b, index));
+      }
+      return getTableCell(b, index).localeCompare(getTableCell(a, index));
+    });
+    setRows(updatedRows);
+  };
+
+  const onRetry = () => {
+    const failedResourceObjects = rows
+      .filter((row) => row.status.error)
+      .map((rowObject) => rowObject.obj);
```

Comment:
This feels backwards. We shouldn't be looking at the table data to find out what's failed. We should track the status of each object in state and generate the table from the status on render.

Reply:
we are actually looking at the status of each object, and generating the table from the status on render, but the naming wasn't indicating that. changed the naming scheme of the state and subsequent methods reacting to state changes